utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -710,9 +710,9 @@ const std::vector<SequenceNumber> WritePreparedTxnDB::GetSnapshotListFromDB(
 
 void WritePreparedTxnDB::ReleaseSnapshotInternal(
     const SequenceNumber snap_seq) {
-  // relax is enough since max increases monotonically, i.e., if snap_seq <
-  // old_max => snap_seq < new_max as well.
-  if (snap_seq < max_evicted_seq_.load(std::memory_order_relaxed)) {
+  // TODO(myabandeh): relax should enough since the synchronizatin is already
+  // done by snapshots_mutex_ under which this function is called.
+  if (snap_seq <= max_evicted_seq_.load(std::memory_order_acquire)) {
```

Comment:
I actually don't follow why we needed to change from relaxed => acquire?

Reply:
The algorithms were initially designed to allow concurrent call to AdvanceMaxEvictedSeq which has added too much unnecessary complexity. We should get rid of all that in a future refactoring.
For the moment my concern was that that if max is already advanced by another thread, and our cpu cache is not updated with that,  then with memory_order_relaxed we might fail to properly cleanup the old_commit_map_ for the released snapshot. I do not think this would happen in the current code but it is easier to prove the code is correct with acquire than with relaxed.